src/Attributable.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace SilverStripe\Assets;
+
+use SilverStripe\Core\Convert;
+
+trait Attributable
+{
+    protected $attributes = [];
+
+//    private static $casting = [
+//        'AttributesHTML' => 'HTMLFragment',
+//        'getAttributesHTML' => 'HTMLFragment',
+//    ];
+
+
+
+    public function setAttribute($name, $value)
+    {
+        $this->attributes[$name] = $value;
+
+        return $this;
+    }
+
+    public function getAttribute($name)
+    {
+        $attributes = $this->getAttributes();
+
+        if (isset($attributes[$name])) {
+            return $attributes[$name];
+        }
+
+        return null;
+    }
+
+    protected abstract function getDefaultAttributes(): array;
```

Comment:
This would require any any class that implements this trait to provide a default list of attributes to get us started.